pkg/storage/tsdb/bucketcache/caching_bucket_config.go

```diff
@@ -12,7 +12,7 @@ import (
 	"github.com/thanos-io/objstore"
 )
 
-// Codec for encoding and decoding results of Iter call.
+// IterCodec defines the interface of a codec used to for encode and decode results of Iter call.
```

Comment:
```suggestion
// IterCodec is a codec for encoding and decoding results of Iter call.
```

(We can already see it's an interface.)